node_io/lib/src/streams.dart

```diff
@@ -186,7 +187,13 @@ class NodeIOSink extends WritableStream<List<int>> implements IOSink {
 
   @override
   void write(Object obj) {
-    _write(encoding.encode(obj.toString()));
+    // Only encode string
+    if (obj is List) {
+      _write(obj);
+    } else {
+      obj = obj.toString();
+      _write(encoding.encode(obj.toString()));
+    }
```

Comment:
According to docs this method is intended to write strings. From IOSink docs:

```dart
/**
   * Converts [obj] to a String by invoking [Object.toString] and
   * [add]s the encoding of the result to the target consumer.
   *
   * This operation is non-blocking. See [flush] or [done] for how to get any
   * errors generated by this call.
   */
```

For byte arrays there is `add()`, afaik.

Reply:
yeap you're right